backend/src/core/loaders/express.loader.ts

```diff
@@ -91,6 +91,13 @@ const expressApp = ({ app }: { app: express.Application }): void => {
   app.use(requestTracerMiddleware)
 
   app.use(overrideContentTypeHeaderMiddleware)
+
+  // We need to parse the body as text specifically for SendGrid callbacks.
+  // This is to avoid stripping whitespace characters and messing with unicode encoding.
+  // This route affects SES callbacks as well, so we'll need to parse the text as JSON
+  // in the parseEvent() handle before parsing the SES event.
+  app.use('/v1/callback/email', bodyParser.text({ type: 'application/json' }))
```

Comment:
SendGrid events contain carriage return and newline characters (`\r\n`) and some even have unicode-encoded characters. The JSON body-parser strips the newline characters and decodes the unicode characters.

I had a less intrusive fix here: https://github.com/opengovsg/postmangovsg/pull/1026/commits/201d0f4729e6b9a032b5f957b7fd0d48d4ee3ecc
But, I think it's not great for reasons stated in the revert: https://github.com/opengovsg/postmangovsg/pull/1026/commits/5473c72307c46a6e1195bda03eead2cbd8d0f28e

I ended up with this solution since it isn't super intrusive, but it does affect the SES callbacks as well.  One other solution is to split the SES and SendGrid callback endpoints.

Reply:
I think this is fine. I've checked and [only one](https://github.com/expressjs/body-parser/blob/480b1cfe29af19c070f4ae96e0d598c099f42a12/lib/types/json.js#L99) `body-parser` middleware will run each time.